### ISSUE TITLE
Switch to aiohttp for outgoing HTTP requests

### ISSUE THREAD

**Issue by yuvipanda** [jupyterhub/oauthenticator#202]
Now that we're on asyncio, we can use the aiohttp library rather than the current curl based one. I believe it'll give us performance improvements generally, especially on PyPy.

**Comment by yuvipanda** [comment]
Ah, I was going to open this in JupyterHub, not OAuthenticator :) Performance is not such a big concern here. Switching would still be nice, probably.

**Comment by manics** [comment]
@yuvipanda do you still plan to do this?

**Comment by yuvipanda** [comment]
No, but someone else should try ;) Will help make JupyterHub be C dependencies free!
